File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> indexes =  new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            indexes[nums[i]] = i;
        }
        for (int j = 0; j < nums.Length; j++) {
            int dif =  target - nums[j];
            if (indexes.ContainsKey(dif) && j != indexes[dif]) {
                return new int[]{j, indexes[dif]};
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        minVal = nums[0]
        l = 0
        r = len(nums) - 1
        m = (l + r) // 2
        if nums[l] < nums[m] < nums[r]:
                return minVal
        while l <= r:
            m = (l + r) // 2
            if (nums[m] > nums[r]):
                if nums[m] < minVal:
                    minVal = nums[m]
                l = m + 1
            else:
                if nums[m] < minVal:
                    minVal = nums[m]
                r = m - 1
        return minVal
            
            

            
            
                
        